bootloader.mk

```diff
@@ -128,7 +128,7 @@ ifeq ($(strip $(BOOTLOADER)), kiibohd)
 endif
 ifeq ($(strip $(BOOTLOADER)), stm32duino)
     OPT_DEFS += -DBOOTLOADER_STM32DUINO
-    MCU_LDSCRIPT = STM32F103x8_stm32duino_bootloader
+    MCU_LDSCRIPT ?= STM32F103x8_stm32duino_bootloader
```

Comment:
This does not seem to work, because `MCU_LDSCRIPT` has already been set by `quantum/mcu_selection.mk` when `bootloader.mk` gets included (the fact that `?=` was used there too apparently does not matter — the first `?=` wins).

Maybe some intermediate variable could be used — e.g., `quantum/mcu_selection.mk` could set `MCU_LDSCRIPT_DEFAULT = something` instead of touching `MCU_LDSCRIPT`, then `bootloader.mk` could override `MCU_LDSCRIPT_DEFAULT` here (again with a simple `=` assignment), and then at some point we could finally have
```make
MCU_LDSCRIPT ?= $(MCU_LDSCRIPT_DEFAULT)
```
Although another problem is that `common_features.mk` is included before `bootloader.mk`, and the EEPROM emulation rules in `common_features.mk` use `MCU_LDSCRIPT`.

One possible workaround is that we could do `MCU_LDSCRIPT ?= $(MCU_LDSCRIPT_DEFAULT)` early (even in `quantum/mcu_selection.mk`), because `MCU_LDSCRIPT` would be recursively expanded if it was set with `?=`, so its value would still reflect any future changes of `MCU_LDSCRIPT_DEFAULT`. Then the code in `common_features.mk` would continue to work as it does now, with a restriction that it won't see the default bootloader-specific script (but would see any keyboard-level override, even if it was for the bootloader-specific script).

Reply:
Just to get this out of my PR list, I've reverted this change; we can figure out what to do later.